File: cppversion/srcs/shellcmd2.cpp
#include "taskmaster.hpp"

void				task_restart(char *input)
{
	std::string temp(&input[8]);
	int			index;
	
	if (temp.empty() != TRUE)
	{
		index = -1;
		while (processes[++index] != NULL)
			if (strcmp(processes[index]->name, &input[8]) == 0)
			{
				processes[index]->restart();
				cout << "TM > User requested 'restart' on program/process " << temp << endl;
				*logFile << currentDateTime() << " User requested 'restart' on program/process " << temp << endl;
			}
	}
	else
		cout << "TM > The 'restart' command requires a program name arguement to function." << endl;
	cout << "TM > ";
}

void				task_kill(char *input)
{
	std::string temp(&input[5]);
	int			index;

	if (temp.empty() != TRUE)
	{
		index = -1;
		while (processes[++index] != NULL)
			if (strcmp(processes[index]->name, &input[5]) == 0)
			{
				processes[index]->kill();
				cout << "TM > User requested 'kill' on program/process " << temp << endl;
				*logFile << currentDateTime() << " User requested 'kill' on program/process " << temp << endl;
			}
	}
	else
		cout << "TM > The 'restart' command requires a program name arguement to function." << endl;
	cout << "TM > ";
}

void				task_shutdown(char *input)
{
	std::string temp(&input[9]);
	int			index;

	if (temp.empty() != TRUE)
	{
		index = -1;
		while (processes[++index] != NULL)
			if (strcmp(processes[index]->name, &input[9]) == 0)
			{
				processes[index]->shutdown();
				cout << "TM > User requested 'shutdown' on program/process " << temp << endl;
				*logFile << currentDateTime() << " User requested 'shutdown' on program/process " << temp << endl;
			}
	}
	else
		cout << "TM > The 'restart' command requires a program name arguement to function." << endl;
	cout << "TM > ";
}

void				task_edit(char *input)
{
	std::string temp(&input[5]);
	int			index;

	if (temp.empty() != TRUE)
	{
		index = -1;
		while (processes[++index] != NULL)
			if (strcmp(processes[index]->name, &input[5]) == 0)
			{
				//processes[index]->edit();
				cout << "TM > You attempted to use the 'edit' command. Unfortunately this command does not yet work." << temp << endl;
				*logFile << currentDateTime() << " User 'edit'ed process <processid>'s <variablename> to <newvalue> " << temp << endl;
			}
	}
	else
		cout << "TM > The 'edit' command requires a program name arguement to function." << endl;
	cout << "TM > ";
}

void				task_start()
{
	std::string temp(&input[9]);
	int			index;

	if (temp.empty() != TRUE)
	{
		index = -1;
		while (processes[++index] != NULL)
			if (strcmp(processes[index]->name, &input[9]) == 0)
			{
				processes[index]->start();
				cout << "TM > User requested 'start' on program/process " << temp << endl;
				*logFile << currentDateTime() << " User requested 'start' on program/process " << temp << endl;
			}
}
	else
		cout << "TM > The 'start' command requires a program name arguement to function." << endl;
	cout << "TM > ";
}

File: phpversion/shellcmd2.php
<?php

function task_restart($input)
{
	$args = explode(" ", $input);
	if (count($args) > 1 && $args[1] != NULL)
	{
		$index = -1;
		while (isset($GLOBALS['processList'][++$index]) == TRUE)
		{
			$process = $GLOBALS['processList'][$index];
			$process->_attribStat['usr_sd'] = FALSE;
			if (strcmp($process->_attribStat['name'], trim($args[1])) == 0)
			{
				$name = trim($args[1]);
				echo "<taskmaster/> User requested 'restart' on program {$name}\n";
				log_message("User requested 'restart' on program {$name}");
				$process->restart();
			}
		}
	}
	else
		echo "<taskmaster/> The 'restart' command requires a program name arguement to function.";
}

function task_kill($input)
{
	$args = explode(" ", $input);
	if (count($args) > 1 && $args[1] != NULL)
	{
		$index = -1;
		while (isset($GLOBALS['processList'][++$index]) == TRUE)
		{
			$process = $GLOBALS['processList'][$index];
			$process->_attribStat['usr_sd'] = TRUE;
			if (strcmp($process->_attribStat['name'], trim($args[1])) == 0)
			{
				$name = trim($args[1]);
				$process->kill();
				echo "<taskmaster/> User requested 'kill' on program {$name}\n";
				log_message("User requested 'kill' on program {$name}");
			}
		}
	}
	else
		echo "<taskmaster/> The 'kill' command requires a program name arguement to function.";
}

function task_shutdown($input)
{
	$args = explode(" ", $input);
	if (count($args) > 1 && $args[1] != NULL)
	{
		$index = -1;
		while (isset($GLOBALS['processList'][++$index]) == TRUE)
		{
			$process = $GLOBALS['processList'][$index];
			$process->_attribStat['usr_sd'] = TRUE;
			if (strcmp($process->_attribStat['name'], trim($args[1])) == 0)
			{
				$process->shutdown();
				$name = trim($args[1]);
				echo "<taskmaster/> User requested 'shutdown' on program {$name}\n";
				log_message("User requested 'shutdown' on program {$name}");
			}
		}
	}
	else
		echo "<taskmaster/> The 'shutdown' command requires a program name arguement to function.";
}

function task_edit($input)
{
	$args = explode(" ", $input);
	if (count($args) > 1 && $args[1] != NULL)
	{
		$index = -1;
		while (isset($GLOBALS['processList'][++$index]) == TRUE)
		{
			$process = $GLOBALS['procesList'][$index];
			if (strcmp($process->_attribStat['name'], trim($args[1])) == 0)
			{
				$name = trim($args[1]);
				//$process->edit();
				echo "<taskmaster/> You attempted to use the 'edit' command. Unfortunately this command does not yet work.{$name}\n";
				log_message("User 'edit'ed process <processid>'s <variablename> to <newvalue> {$name}");
			}
		}
	}
	else
		echo "<taskmaster/> The 'edit' command requires a program name arguement to function.";
}

function task_start($input)
{
	$args = explode(" ", $input);
	if (count($args) > 1 && $args[1] != NULL)
	{
		$index = -1;
		while (isset($GLOBALS['processList'][++$index]) == TRUE)
		{
			$process = $GLOBALS['processList'][$index];
			if (strcmp($process->_attribStat['name'], trim($args[1])) == 0)
			{
				$name = trim($args[1]);
				$process->_attribStat['usr_sd'] = FALSE;
				$process->start();
				echo "<taskmaster/> User requested 'start' on program {$name}\n";
				log_message("User requested 'start' on program {$name}");
			}
		}
	}
	else
		echo "<taskmaster/> The 'start' command requires a program name arguement to function.";
}
?>